Employee_Management_Portal/EmployeeManagementPortal/EmployeeManagementPortal/src/main/java/com/portal/service/EmployeeService.java

```diff
@@ -0,0 +1,457 @@
+     * @return An object representing the employee with
+     * the specified email address,or null if not found.
+     */
+    public final EmployeeOutDTO getEmployeeByEmail(final String email) {
+        LOGGER.error("Employee id already exists");
+        Employee employee = userRepository.findByEmail(email).get();
+        EmployeeOutDTO empDto = new EmployeeOutDTO();
+        if (employee != null) {
+            empDto.setId(employee.getId());
+            empDto.setName(employee.getName());
+            empDto.setEmail(employee.getEmail());
+            empDto.setEmpId(employee.getEmpId());
+            empDto.setDesignation(employee.getDesignation());
+            empDto.setContactNumber(employee.getContactNumber());
+            empDto.setDob(employee.getDob());
+            empDto.setDoj(employee.getDoj());
+            empDto.setLocation(employee.getLocation());
+            empDto.setRole(employee.getRole());
+            empDto.setManagerId(employee.getManagerId());
+            empDto.setProjectId(employee.getProjectId());
+            Employee manager = userRepository
+                    .findById(employee.getManagerId()).get();
+            empDto.setManager(manager.getName());
+            empDto.setSkills(employee.getSkills());
+
+            if (employee.getProjectId() == 0) {
+                empDto.setProjectName("");
+            } else {
+                Project project = projectRepository
+                        .findById(employee.getProjectId()).get();
+                empDto.setProjectName(project.getName());
+            }
+
+//                Employee manager = userRepository
+//                        .findById(employee.getManagerId()).get();
+//                empDto.setManagerId(manager.getName());
+//                empDto.setSkills(null);
+        }
+        return empDto;
+    }
+
+    /**
+     * update project to employee.
+     * @param id        employee primary key id
+     * @param projectId project id
+     * @param managerId manager id
+     * @return general response dto
+     */
+    public final ApiResponseDTO updatedProject(final Long id,
+            final Long projectId, final Long managerId) {
+
+        Employee employee = userRepository.findById(id).orElse(null);
+        if (employee != null) {
+
+            employee.setProjectId(projectId);
+            employee.setManagerId(managerId);
+            userRepository.save(employee);
+            List<RequestResource> request = requestResourceRepository
+                    .findByEmployeeId(id);
+            for (RequestResource req : request) {
+                requestResourceService.rejectRequest(req.getResourceId());
+            }
+            ApiResponseDTO response = new ApiResponseDTO();
+            response.setMessage("Updated Suceesfully");
+            return response;
+        }
+        LOGGER.error("Employee not found");
+        throw new ResourceNotFoundException("Employee not found");
```

Comment:
do not pass string message directly when throwing exceptions. Make another class ('ex: ErrorConstants.java) and store all messages in that class as constants and then call it here.